speculos/main.py

```diff
@@ -46,6 +46,28 @@ def set_pdeath(sig):
     libc.prctl(PR_SET_PDEATHSIG, sig)
 
 
+ELF_METADATA_SECTIONS = [
+    "target",
+    "app_name",
+    "app_version",
+    "api_level"
+]
+
+
+def get_elf_ledger_metadata(app_path):
+    with open(app_path, 'rb') as fp:
+        elf = ELFFile(fp)
+        metadata = {}
+        for section_name in ELF_METADATA_SECTIONS:
+            section = elf.get_section_by_name(f"ledger.{section_name}")
+            if section:
+                metadata[section_name] = section.data().decode("utf-8").strip()
+                if section_name == "target":
+                    if metadata[section_name] == "nanos2":
+                        metadata[section_name] = "nanosp"
```

Comment:
why not just change the nanosp to nanos2 in speculos?

Reply:
Not a fight for today :)